src/pages/Lend/LoanSearchPage.vue

```diff
@@ -301,14 +317,25 @@ export default {
 				teamId = teamInfo?.community?.team?.id ?? null;
 			} else {
 				const userTeamsData = this.apollo.readQuery({ query: myTeamsQuery });
+				const goalsData = this.apollo.readQuery({ query: teamsGoalsQuery, variables: { isActive: true, status: 'IN_PROGRESS' } }); // eslint-disable-line max-len
 				const userTeams = userTeamsData.my?.teams?.values ?? [];
+				const activeGoals = goalsData?.goals?.values ?? [];
 
-				let maxAmountLentTeam = {};
+				let sortedAmountLentTeams = [];
+				let userChallengeTeam = {};
 				if (userTeams.length > 0) {
-					maxAmountLentTeam = reduceAmountLentTeamArray(userTeams);
+					sortedAmountLentTeams = sortAmountLentTeamArray(userTeams);
+					for (let index = 0; index < sortedAmountLentTeams.length; index += 1) {
```

Comment:
This duplicate code can be moved into a shared method, like we have for `sortAmountLentTeamArray`. It looks like this version is slightly safer than the one above, since it checks `goal` as well on line 331.

Reply:
Oh yeah, I missed that check there...
Sounds good having a shared method, let's refactor this.

Reply:
Done!